### PR TITLE
Correct fee calculations to match legacy Dogecoin

### PR DESCRIPTION
**Pull request by rnicoll**

### CODE REVIEW

**Comment by langerhans** [comment]
There is block here: https://github.com/dogecoin/dogecoin/blob/1.10-dev/src/main.cpp#L1003-L1008  
That leads to rejection of 0-fee-txns. The problem with that is, that old clients will still create 0 fee transactions if told to. I see that in my logs and current 1.10 rejects them. As much as I'd like to keep it that way, it could turn out very inconvenient for those people... So a change in that bit is probably needed, maybe in the bits below too (?)

File: src/main.cpp [hunk 1]
```diff
@@ -872,8 +872,7 @@ CAmount GetMinRelayFee(const CTransaction& tx, unsigned int nBytes, bool fAllowF
     CAmount nMinFee = ::minRelayTxFee.GetFee(nBytes);
     nMinFee += GetDogecoinDustFee(tx.vout, ::minRelayTxFee);
 
-    // Dogecoin: Disable free transactions
-    /* if (fAllowFree)
+    if (fAllowFree)
```
Comment (rnicoll) [inline src/main.cpp]: I've re-enabled free transaction relaying here, which should match behaviour in 1.8

**Comment by rnicoll** [comment]
Updated!

**Comment by patricklodder** [comment]
seems to break an IsStandard() check? see https://travis-ci.org/dogecoin/dogecoin/jobs/78713609#L1265

**Comment by rnicoll** [comment]
Removed the test, it's looking for dust transactions to be marked as non-standard, but as we allow them, we consider them standard. Haven't squashed, won't be around for much time until Sunday, so if you want it squashed you'll have to wait until then :)

**Comment by patricklodder** [comment]
Missing the IsStandard() fix after rebase.

**Comment by rnicoll** [comment]
Re-added IsStandard() fix

**Comment by langerhans** [comment]
Played around with this and seems like it does what it says :)

**Pull request merged**